votes/0639-propose-a-release-note-guide.md

```diff
@@ -0,0 +1,76 @@
+# Propose a Release Notes Language Style Guide
+
+## Background
+
+A concise release note can clearly and accurately deliver to users how your PR can make a difference. Your release note written in a PR will be presented in [docs.pingcap.com](https://docs.pingcap.com/tidb/stable/release-notes) as a part of the TiDB documentation.
+
+I hereby propose a release notes language style guide that briefly explains what a quality release note looks like, provides examples, and aims to help you write quality release notes.
+
+## What a quality release note looks like
+
+A high-quality release note has the following merits:
+
+- Clear in type
+- Adequate and clear in meaning
+- User perspective
+
+A release note with a distinguishable type can help users quickly identify the nature or goal of your PR change. Other teams will also benefit from it.
+
+Depending on what your PR changes, you can refer to one of the following release note types:
+
+- Compatibility change
+- Improvement or Feature enhancement
+- Bug fix
+
+## Compatibility change
+
+A compatibility change note means:
+
+- Your PR adds, removes, or modifies one or more configuration items or system variables.
+- Your PR modifies the default value of a configuration item or system variable.
+
+For this type of note, you should clearly and adequately state the following aspects:
+
+- The previous code behavior, configuration item, or default value.
+- The new code behavior, configuration item, or default value since the new version.
+
+Note that the object of the change should be user-perceivable. If the changed configuration item or system variable is not supposed to be exposed to users, do not include it in your release notes.
+
+**Examples:**
+
+| **Not recommended**                                          | **Clear in type** | **Adequate and clear in meaning** | **User perspective** | Recommended                                                  |
+| ------------------------------------------------------------ | ----------------- | --------------------------------- | -------------------- | ------------------------------------------------------------ |
+| copr: cast invalid utf8 string to real bug fix               | ❌                 | ❌                                 | ❌                    | Before v4.0.16, when TiDB converts an illegal UTF-8 string to a Real type, an error is reported directly. Starting from v4.0.16, TiDB processes the conversion according to the legal UTF-8 prefix in the string |
```

Comment:
I wonder if it is a good idea to mention version number in release note. Consider the PR (containing release note) can cherry-pick to different release branches, we will have to manually update the version number.

Reply:
Each release notes file is dedicated to a specific version, for example, v4.0.16. Even if this file is cherry-picked to other branches, it still only applies to that specific version (v4.0.16). For preparation, each release notes file is prepared separately, and the manual labor to change version number is almost negligible compared with the total effort required to prepare the entire release notes for each release.

Reply:
I thought release notes are written in PR description and we have some tool to collect notes and generate full release note for each version.

Reply:
Yes, you are right. That's exactly how we are collecting release notes from the PR description part. The problem is that the collected raw material might not always be usable or deliverable. That's why we propose this release note guide. The aim is to help developers write more usable or deliverable notes.

Reply:
The problem is that when we write release note in a PR. We cannot know what version will contain the patch and what branches the PR will be cherry-picked.

Reply:
I see. I'll remove the version number from the example. PTAL whether this modification is OK. Thanks~
```suggestion
| copr: cast invalid utf8 string to real bug fix               | ❌                 | ❌                                 | ❌                    | Previously, when TiDB converts an illegal UTF-8 string to a Real type, an error is reported directly. From now on, TiDB will process the conversion according to the legal UTF-8 prefix in the string. |
```

Reply:
I think this release note guide seems to be more useful when it is applied to the process of writing a release note for a release by PMs, **instead of to be used in PRs**, for these reasons:

1. Multiple PRs may contribute to the same product behavior. For example, improve performance, improve performance, ... . It is not necessary to expand the details for each one.
2. Most of the developers will not get it right to be user perspective. Developers are not PMs.
3. A friendly release note contains much more than what developers care about. For example, previous behaviors, changed versions (mentioned by @disksing in this thread), etc. Developers is very likely to not help you do it as it is out of responsibility.

I mean, maybe our PMs should all read this, and strictly follow this :) Instead of devs. However it is always good to have some attempts. Keep going!